gammapy/spectrum/utils.py

```diff
@@ -13,19 +16,254 @@
+
+        if not isinstance(bin_edges, Quantity) and not isinstance(bin_centers, Quantity):
+            raise ValueError("Energies must be Quantity objects")
+
+        self._bin_edges = bin_edges
+        self._bin_centers = bin_centers
+
+        if bin_centers is not None:
+            self.nbins = len(self.bin_centers)
+        elif bin_edges is not None:
+            self.nbins = len(self.bin_edges) - 1
+        else:
+            self.nbins = 0
+
+    @staticmethod
+    def from_edges(bin_edges, centers='log'):
+        """Create EnergyBinning by given the bin edges and a method how to place the bin centers.
+
+        Options for placing the bin centers are:
+
+        - 'log' : Place bin centers at the log center of the bin (default)
+        - to be implemented
+
+        Parameters
+        ----------
+        bin_edges : `~astropy.units.Quantity`
+        Energy bin edges
+        centeres : str
+        Method how to place the energy bin centers
+
+        Returns
+        -------
+        Energy Binning
+        """
+        if centers == 'log':
+            bin_centers = find_log_centers(bin_edges)
+        else:
+            raise ValueError("Method for placing bin centers not implemented")
+
+        return EnergyBinning(bin_edges, bin_centers)
+
+    @staticmethod
+    def from_centers(bin_centers):
+        """Create EnergyBinning by given the bin centers
+
+        The difference between this method and the contrsuctor is that a check for equal log spacing is done. If the given bin centers are equally spaced in log, the bin edges are computed  (I don't know if thats really necessary, it doesn't hurt for sure)
+        Parameters
+        ----------
+        bin_centers : `~astropy.units.Quantity`
+        Energy bin centers
+
+        Returns
+        -------
+        Energy Binning
+        """
+
+        # TODO: test for equal log spacing
+        bin_edges = find_log_edges(bin_centers)
+
+        return EnergyBinning(bin_edges, bin_centers)
+
+    @staticmethod
+    def equal_log_spacing(emin, emax, nbins):
+        """Create EnergyBinning with equal log-spacing by giving the outer bin edges
+
+        Parameters
+        ----------
+        emin : `~astropy.units.Quantity`
+        Lower bound of lowest energy bin
+        emax : `~astropy.units.Quantity`
+        Upper bound of highest energy bin
+        bins : int
+        Number of bins
+
+        Returns
+        -------
+        Energy Binning
+        """
+
+        if not isinstance(emin, Quantity) or not isinstance(emax, Quantity):
+            raise ValueError("Energies must be Quantity objects.")
+
+        emin = emin.to('TeV')
+        emax = emax.to('TeV')
+
+        x_min, x_max = np.log10([emin.value, emax.value])
+        bin_edges = np.logspace(x_min, x_max, nbins + 1)
+
+        bin_edges = Quantity(bin_edges, emin.unit)
+
+        return EnergyBinning.from_edges(bin_edges, 'log')
+
+    @staticmethod
+    def from_fits(hdulist):
```

Comment:
I prefer separate methods for the separate formats.
It means the caller has to be more explicit, but that's a good thing.